core/src/main/java/com/cloud/agent/resource/virtualnetwork/VirtualRoutingResource.java

```diff
@@ -268,6 +267,53 @@ private CheckS2SVpnConnectionsAnswer execute(CheckS2SVpnConnectionsCommand cmd)
         return new CheckS2SVpnConnectionsAnswer(cmd, result.isSuccess(), result.getDetails());
     }
 
+    private GetRouterMonitorResultsAnswer execute(GetRouterMonitorResultsCommand cmd) {
+
+        String routerIp = cmd.getAccessDetail(NetworkElementCommand.ROUTER_IP);
+        String args = cmd.shouldPerformFreshChecks() ? "true" : "false";
+        s_logger.debug("Fetching health check result for " + routerIp + " and executing fresh checks: " + args);
+        ExecutionResult result = _vrDeployer.executeInVR(routerIp, VRScripts.ROUTER_MONITOR_RESULTS, args);
+        if (result.isSuccess()) {
+            List<String> failingChecks = new ArrayList<>();
+            StringBuilder monitorResults = new StringBuilder();
+            if (!result.getDetails().isEmpty()) {
+                String[] lines = result.getDetails().trim().split("\n");
```

Comment:
To increase readability, can you split this into multiple methods?

Reply:
I've refactored to increase readability.